diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
index 0437f3a4..93153e97 100644
--- a/python-package/xgboost/libpath.py
+++ b/python-package/xgboost/libpath.py
@@ -29,7 +29,10 @@ def find_lib_path() -> List[str]:
         # option.
         os.path.join(sys.base_prefix, "lib"),
     ]
-
+    if sys.platform == 'win32':
+        dll_path = [os.path.join(sys.prefix, 'Library', 'mingw-w64', 'bin')]
+    else:
+        dll_path = [os.path.join(sys.prefix, 'lib')]
     if sys.platform == "win32":
         if platform.architecture()[0] == "64bit":
             dll_path.append(os.path.join(curr_path, "../../windows/x64/Release/"))
