diff a/dmlc-core/include/dmlc/base.h b/dmlc-core/include/dmlc/base.h
index 085ca61..6b84167 100644
--- a/dmlc-core/include/dmlc/base.h
+++ b/dmlc-core/include/dmlc/base.h
@@ -161,9 +161,6 @@
 #if !defined(__GNUC__)
 #define fopen64 std::fopen
 #endif
-#if (defined __MINGW32__) && !(defined __MINGW64__)
-#define fopen64 std::fopen
-#endif
 #ifdef _MSC_VER
 #if _MSC_VER < 1900
 // NOTE: sprintf_s is not equivalent to snprintf,
