diff --git a/R-package/src/Makevars.in b/R-package/src/Makevars.in
index 65a26bbb..d28e9fb3 100644
--- a/R-package/src/Makevars.in
+++ b/R-package/src/Makevars.in
@@ -1,5 +1,5 @@
 # package root
-PKGROOT=../../
+PKGROOT=../..
 ENABLE_STD_THREAD=1
 # _*_ mode: Makefile; _*_
 
diff --git a/R-package/src/Makevars.win b/R-package/src/Makevars.win
index 3583b4f1..e057e1a4 100644
--- a/R-package/src/Makevars.win
+++ b/R-package/src/Makevars.win
@@ -1,5 +1,5 @@
 # package root
-PKGROOT=./
+PKGROOT=../..
 ENABLE_STD_THREAD=0
 # _*_ mode: Makefile; _*_
 
