From 3660a4216763d02d48cb1db63f5fee7ad02ad7d4 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Fri, 28 Jun 2024 17:38:34 -0700
Subject: [PATCH] [PATCH] Undo dmlc/xgboost#9436

---
 CMakeLists.txt    | 9 ---------
 cmake/Utils.cmake | 1 -
 2 files changed, 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4ca82937..dc0f7e654 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -301,15 +301,6 @@ add_subdirectory(${xgboost_SOURCE_DIR}/plugin)
 
 if(PLUGIN_RMM)
   find_package(rmm REQUIRED)
-
-  # Patch the rmm targets so they reference the static cudart
-  # Remove this patch once RMM stops specifying cudart requirement
-  # (since RMM is a header-only library, it should not specify cudart in its CMake config)
-  get_target_property(rmm_link_libs rmm::rmm INTERFACE_LINK_LIBRARIES)
-  list(REMOVE_ITEM rmm_link_libs CUDA::cudart)
-  list(APPEND rmm_link_libs CUDA::cudart_static)
-  set_target_properties(rmm::rmm PROPERTIES INTERFACE_LINK_LIBRARIES "${rmm_link_libs}")
-  get_target_property(rmm_link_libs rmm::rmm INTERFACE_LINK_LIBRARIES)
 endif()
 
 if(PLUGIN_SYCL)
diff --git a/cmake/Utils.cmake b/cmake/Utils.cmake
index d555f5edf..8659df7c3 100644
--- a/cmake/Utils.cmake
+++ b/cmake/Utils.cmake
@@ -240,7 +240,6 @@ macro(xgboost_target_link_libraries target)
 
   if(USE_CUDA)
     xgboost_set_cuda_flags(${target})
-    target_link_libraries(${target} PUBLIC CUDA::cudart_static)
   endif()
 
   if(PLUGIN_RMM)
-- 
2.34.1

