From 4165b0eb2665d27142d005b5a062ce6d84fe0432 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <chohyu01@cs.washington.edu>
Date: Mon, 22 Jul 2024 12:20:10 -0700
Subject: [PATCH] [CMake] Explicitly link with CCCL (standalone or CTK)

---
 CMakeLists.txt    | 6 ++++++
 cmake/Utils.cmake | 1 +
 2 files changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc0f7e654..8371a45c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,6 +224,12 @@ if(USE_CUDA)
   add_subdirectory(${PROJECT_SOURCE_DIR}/gputreeshap)
 
   find_package(CUDAToolkit REQUIRED)
+  find_package(CCCL CONFIG)
+  if(NOT CCCL_FOUND)
+    message(STATUS "Standalone CCCL not found. Attempting to use CCCL from CTK...")
+    find_package(CCCL REQUIRED CONFIG
+      PATHS ${CUDAToolkit_LIBRARY_DIR}/cmake/cccl)
+  endif()
 endif()
 
 if(FORCE_COLORED_OUTPUT AND (CMAKE_GENERATOR STREQUAL "Ninja") AND
@@ -422,6 +428,9 @@ else()
     set(INSTALL_TARGETS xgboost)
   endif()
 endif()
 
 install(TARGETS ${INSTALL_TARGETS}
   EXPORT XGBoostTargets
diff --git a/cmake/Utils.cmake b/cmake/Utils.cmake
index 8659df7c3..01dca611c 100644
--- a/cmake/Utils.cmake
+++ b/cmake/Utils.cmake
@@ -108,6 +108,7 @@ function(xgboost_set_cuda_flags target)
     target_compile_definitions(${target} PRIVATE -DXGBOOST_USE_NVTX=1)
   endif()
 
+  target_link_libraries(${target} PRIVATE CCCL::CCCL)
   target_compile_definitions(${target} PRIVATE -DXGBOOST_USE_CUDA=1)
   target_include_directories(
     ${target} PRIVATE
-- 
2.34.1

