From 7b5a3d0347abc947c2d19730bf38f5460b4881ad Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Fri, 7 Aug 2026 23:28:19 +0800
Subject: [PATCH] Enable latest libcxx on MacOS.

---
 R-package/src/Makevars.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/R-package/src/Makevars.in b/R-package/src/Makevars.in
index afc1bff67..f02dd7a93 100644
--- a/R-package/src/Makevars.in
+++ b/R-package/src/Makevars.in
@@ -13,7 +13,8 @@ XGB_RFLAGS = \
     -DDMLC_LOG_BEFORE_THROW=0 \
     -DDMLC_ENABLE_STD_THREAD=$(ENABLE_STD_THREAD) \
     -DDMLC_DISABLE_STDIN=1 \
-    -DDMLC_LOG_CUSTOMIZE=1
+    -DDMLC_LOG_CUSTOMIZE=1 \
+    -D_LIBCPP_DISABLE_AVAILABILITY
 $(foreach v, $(XGB_RFLAGS), $(warning $(v)))

 PKG_CPPFLAGS = \
--
2.53.0
