From a05799f60e1174277d467430e699ba2ce424aebf Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Mon, 12 Jan 2026 22:37:59 -0800
Subject: [PATCH 2/5] Remove nvidia-nccl-cu12 from pyproject.toml

---
 python-package/pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/python-package/pyproject.toml b/python-package/pyproject.toml
index 934c1da6b..5db2e7809 100644
--- a/python-package/pyproject.toml
+++ b/python-package/pyproject.toml
@@ -34,7 +34,6 @@ classifiers = [
 dependencies = [
     "numpy",
     "scipy",
-    "nvidia-nccl-cu12 ; platform_system == 'Linux'",
 ]
 
 [project.urls]
-- 
2.54.0

