From bc7bab8c97c66de3885846866337077731c5cda9 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Wed, 27 Nov 2024 00:51:35 -0800
Subject: [PATCH] Remove nvidia-nccl-cu12 from pyproject.toml

---
 python-package/pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/python-package/pyproject.toml b/python-package/pyproject.toml
index 158d900d9..aad21f3ca 100644
--- a/python-package/pyproject.toml
+++ b/python-package/pyproject.toml
@@ -33,7 +33,6 @@ classifiers = [
 dependencies = [
     "numpy",
     "scipy",
-    "nvidia-nccl-cu12 ; platform_system == 'Linux' and platform_machine != 'aarch64'",
 ]
 
 [project.urls]
-- 
2.48.1

