From e2c4aa25cf84ad7db68748ca7d0c8275e5dcef91 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Wed, 27 Nov 2024 00:51:35 -0800
Subject: [PATCH] Remove nvidia-nccl-cu12 from pyproject.toml

---
 python-package/pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/python-package/pyproject.toml b/python-package/pyproject.toml
index e6cfde49d..0201c8dea 100644
--- a/python-package/pyproject.toml
+++ b/python-package/pyproject.toml
@@ -31,7 +31,6 @@ classifiers = [
 dependencies = [
     "numpy",
     "scipy",
-    "nvidia-nccl-cu12 ; platform_system == 'Linux' and platform_machine != 'aarch64'",
 ]
 
 [project.urls]
-- 
2.47.0

