From 3cff598144b2843b21eeccb42075f47c84127bb7 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Sat, 28 Sep 2024 21:12:02 +0200
Subject: [PATCH 3/3] Mark wheels as any platform compatible

---
 python-package/hatch_build.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/python-package/hatch_build.py b/python-package/hatch_build.py
index 925c917..710530c 100644
--- a/python-package/hatch_build.py
+++ b/python-package/hatch_build.py
@@ -3,7 +3,6 @@ Custom hook to customize the behavior of Hatchling.
 Here, we customize the tag of the generated wheels.
 """
 
-import sysconfig
 from typing import Any, Dict
 
 from hatchling.builders.hooks.plugin.interface import BuildHookInterface
@@ -11,8 +10,7 @@ from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 
 def get_tag() -> str:
     """Get appropriate wheel tag according to system"""
-    tag_platform = sysconfig.get_platform().replace("-", "_").replace(".", "_")
-    return f"py3-none-{tag_platform}"
+    return f"py3-none-any"
 
 
 class CustomBuildHook(BuildHookInterface):
