From 41b851d086176037355faca6695e1d53f0239e19 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Sun, 25 May 2025 02:18:12 -0700
Subject: [PATCH] Disable glibc version check

---
 python-package/xgboost/core.py | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/python-package/xgboost/core.py b/python-package/xgboost/core.py
index 6caab146e..7356f3357 100644
--- a/python-package/xgboost/core.py
+++ b/python-package/xgboost/core.py
@@ -379,30 +379,6 @@ def build_info() -> dict:
     return res
 
 
-def _check_glibc() -> None:
-    if is_sphinx_build():
-        return
-
-    glibc_ver = build_info().get("GLIBC_VERSION", None)
-    if glibc_ver is not None and (
-        glibc_ver[0] < 2 or glibc_ver[0] == 2 and glibc_ver[1] < 28
-    ):
-        warnings.warn(
-            "Your system has an old version of glibc (< 2.28). We will stop supporting "
-            "Linux distros with glibc older than 2.28 after **May 31, 2025**. "
-            "Please upgrade to a recent Linux distro (with glibc >= 2.28) to use "
-            "future versions of XGBoost.\n"
-            "Note: You have installed the 'manylinux2014' variant of XGBoost. Certain "
-            "features such as GPU algorithms or federated learning are not available. "
-            "To use these features, please upgrade to a recent Linux distro with glibc "
-            "2.28+, and install the 'manylinux_2_28' variant.",
-            FutureWarning,
-        )
-
-
-_check_glibc()
-
-
 def _numpy2ctypes_type(dtype: Type[np.number]) -> Type[CNumeric]:
     _NUMPY_TO_CTYPES_MAPPING: Dict[Type[np.number], Type[CNumeric]] = {
         np.float32: ctypes.c_float,
-- 
2.48.1

